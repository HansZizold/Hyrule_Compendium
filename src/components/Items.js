import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';

import { getItems } from '../redux/items/items';
import Category from './Category';
import styles from './Items.module.css';

const Items = () => {
  const itemlist = useSelector((state) => state.items);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getItems('equipment'));
  }, [dispatch]);

  return (
    <div>
      <Category />
      <div className={styles.mainitems}>
        {itemlist.map((elem) => {
          const {
            id, name, image, attack, defense, cookingeffect, commonlocations, category,
          } = elem;
          let locations = 'Unavailable';
          if (commonlocations) [locations] = commonlocations;
          if (category === 'equipment') {
            return (
              <Link
                to={`${name}/${id}`}
                key={id}
                className={styles.singleitem}
              >
                <h3 className={styles.itemname}>{name}</h3>
                <img src={image} alt="Hyrule Compendium" width="300" />
                <p className={styles.features}>
                  Attack:
                  {' '}
                  {attack || '0'}
                  {' '}
                  Defense:
                  {' '}
                  {defense || '0'}
                </p>
              </Link>
            );
          }

          if (category === 'creatures' || category === 'materials') {
            return (
              <div key={id} className={styles.singleitem}>
                <h3 className={styles.itemname}>{name}</h3>
                <img src={image} alt="Hyrule Compendium" width="300" />
                <p className={styles.cooking}>
                  Cooking Effect:
                  {' '}
                  {cookingeffect || 'No Effect'}
                </p>
              </div>
            );
          }

          if (category === 'monsters' || category === 'treasure') {
            return (
              <div key={id} className={styles.singleitem}>
                <h3 className={styles.itemname}>{name}</h3>
                <img src={image} alt="Hyrule Compendium" width="300" />
                <p className={styles.cooking}>
                  Common Location:
                  {' '}
                  {locations}
                </p>
              </div>
            );
          }

          return null;
        })}
      </div>
    </div>
  );
};
export default Items;
