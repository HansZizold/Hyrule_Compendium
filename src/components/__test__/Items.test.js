import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Items from '../Items';
import store from '../../redux/configureStore';

describe(' Items Component Rendering ', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Items />
        </Provider>,
      );
    expect(tree).toMatchSnapshot();
  });
});
