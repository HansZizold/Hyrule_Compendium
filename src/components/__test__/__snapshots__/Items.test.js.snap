// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Items Component Rendering  renders correctly 1`] = `
<div>
  <div
    className="category"
  >
    <h2
      className="select"
    >
      SELECT CATEGORY
    </h2>
    <select
      className="options"
      id="category"
      name="category"
      onChange={[Function]}
    >
      <option
        className="singleoption"
        value=""
      >
        Choose a Category
      </option>
      <option
        className="singleoption"
        value="equipment"
      >
        Equipment
      </option>
      <option
        className="singleoption"
        value="creatures"
      >
        Creatures
      </option>
      <option
        className="singleoption"
        value="materials"
      >
        Materials
      </option>
      <option
        className="singleoption"
        value="monsters"
      >
        Monsters
      </option>
      <option
        className="singleoption"
        value="treasure"
      >
        Treasure
      </option>
    </select>
  </div>
  <div
    className="mainitems"
  />
</div>
`;
